sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -328,6 +324,23 @@ object RapidsExecutorPlugin {
       zipped.forall { case (e, a) => e <= a }
     }
   }
+
+  private val CUDA_EXCEPTION: String = classOf[ai.rapids.cudf.CudaException].getName
+  private val CUDA_FATAL_EXCEPTION: String = classOf[ai.rapids.cudf.CudaFatalException].getName
+  private val CUDF_EXCEPTION: String = classOf[ai.rapids.cudf.CudfException].getName
+
+  def isCudaException(ef: ExceptionFailure): Boolean = {
+    ef.description.contains(CUDA_EXCEPTION) || ef.toErrorString.contains(CUDA_EXCEPTION)
```

Comment:
are the cudf exceptions not serializable?  Otherwise it looks like we should be able to use exception() function call to look at the actual exception type.   I guess we can keep the string parsing if the exception is None.  @jlowe for opinion

Reply:
Agree we should pursue a solution that does not rely on scraping description messages if possible.

Reply:
Done.